lib/MusicBrainz/Server/Controller/Role/Tag.pm

```diff
@@ -33,13 +33,33 @@ sub tags : Chained('load') PathPart('tags') {
     my @tags = $c->model($self->{model})->tags->find_tags($entity->id);
     my @display_tags = grep { $_->count > 0 && !$_->tag->is_genre_tag } @tags;
     my @display_genres = grep { $_->count > 0 && $_->tag->is_genre_tag } @tags;
-
-    $c->stash(
-        display_tags => \@display_tags,
-        display_genres => \@display_genres,
-        tags_json => $c->json->encode(\@tags),
-        template => 'entity/tags.tt',
-    );
+    my $tags_model = $c->model($self->{model})->tags;
+    my @user_tags = $tags_model->find_user_tags($c->user->id, $entity->id)
+        if $c->user_exists;
+    my $count = $tags_model->find_tag_count($entity->id);
+    my @toptags = $tags_model->find_top_tags($entity->id, $TOP_TAGS_COUNT);
```

Comment:
(Sorry, missed this before.) All(?) the lines you added here are already done in `load` above - you can just reference `$c->stash->{top_tags}`, `$c->stash->{user_tags}`, etc. to get the same data.

Reply:
Should be done now :)